docs/devguide/testing.asciidoc

```diff
@@ -0,0 +1,67 @@
+[[testing]]
+=== Testing
+
+Beats has a various sets of tests. This guide should help to understand how the different test suites work, how they are used and new tests are added.
+
+In general there are two major test suites:
+
+* Tests written in Golang
+* Tests written in Python
+
+The tests written in Golang use the https://golang.org/pkg/testing/[Golang Testing package]. The tests written in Python depend on http://nose.readthedocs.io/en/latest/[nosetests] and required a compiled and executable binary from the Golang code. The python test run a beat with a specific config and params and either check if the output is as expected or if the correct things show up in the logs.
+
+For both of the above test suites so called integration tests exists. Integration tests in Beats are tests which require an external system like Elasticsearch to test if the integration with this service works as expected. Beats provides in its testsuite docker containers and docker-compose files to start these environments but a developer can run the required services also locally.
```

Comment:
> Integration tests in Beats are tests which require an external system

I thought *Integration tests*  are long running tests that *can* depend on external systems. If you run integration tests without setting an `ENV` only tests not relying on external system run, otherwise all. 
But maybe I misunderstood this?